packages/ember/tests/component_context_test.js

```diff
@@ -0,0 +1,145 @@
+    });
+  }
+
+  ['@test properties of a component without a template should not collide with internal structures [DEPRECATED]'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{my-component data=foo}}</div>');
+
+    this.add('controller:application', Controller.extend({
+      'text': 'outer',
+      'foo': 'Some text inserted by jQuery'
+    }));
+    this.addComponent('my-component', { ComponentClass: Component.extend({
+      didInsertElement() {
+        this.$().html(this.get('data'));
+      }
+    }) });
+
+    this.visit('/').then(() => {
+      assert.equal(jQuery('#wrapper').text(), 'Some text inserted by jQuery', 'The component is composed correctly');
+    });
+  }
+
+  ['@test attrs property of a component without a template should not collide with internal structures'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{my-component attrs=foo}}</div>');
+
+    this.add('controller:application', Controller.extend({
+      'text': 'outer',
+      'foo': 'Some text inserted by jQuery'
+    }));
+    this.addComponent('my-component', { ComponentClass: Component.extend({
+      didInsertElement() {
+        // FIXME: I'm unsure if this is even the right way to access attrs
+        this.$().html(this.get('attrs.attrs.value'));
+      }
+    }) });
+
+    this.visit('/').then(() => {
+      assert.equal(jQuery('#wrapper').text(), 'Some text inserted by jQuery', 'The component is composed correctly');
+    });
+  }
+
+  ['@test Components trigger actions in the parents context when called from within a block'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}<a href=\'#\' id=\'fizzbuzz\' {{action \'fizzbuzz\'}}>Fizzbuzz</a>{{/my-component}}</div>');
+
+    this.add('controller:application', Controller.extend({
+      actions: {
+        fizzbuzz() {
+          ok(true, 'action triggered on parent');
+        }
+      }
+    }));
+    this.addComponent('my-component', { ComponentClass: Component.extend({}) });
+
+    this.visit('/').then(() => {
+      run(() => jQuery('#fizzbuzz', '#wrapper').click());
+    });
+  }
+
+  ['@test Components trigger actions in the components context when called from within its template'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
+
+    this.add('controller:application', Controller.extend({
+      actions: {
+        fizzbuzz() {
+          ok(false, 'action on the wrong context');
+        }
+      }
+    }));
+    this.addComponent('my-component', { ComponentClass: Component.extend({
+      actions: {
+        fizzbuzz() {
+          ok(true, 'action triggered on component');
+        }
+      }
+    }), template: '<a href=\'#\' id=\'fizzbuzz\' {{action \'fizzbuzz\'}}>Fizzbuzz</a>' });
+
+    this.visit('/').then(() => {
+      run(() => jQuery('#fizzbuzz', '#wrapper').click());
```

Comment:
runloop should not be needed here, or around any other `click` statements. `this.$('#fizzbuzz').click()` would be fine.